font-codegen/src/flags_enums.rs

```diff
@@ -321,25 +321,30 @@ pub(crate) fn generate_raw_enum(raw: &RawEnum) -> TokenStream {
     let variants = raw.variants.iter().map(|variant| {
         let name = &variant.name;
         let value = &variant.value;
-        let docs = &variant.docs;
+        let docs = &variant.attrs.docs;
+        let maybe_default = variant.attrs.default.then(|| quote!(#[default]));
         quote! {
             #( #docs )*
+            #maybe_default
             #name = #value,
         }
     });
+
     let variant_inits = raw.variants.iter().map(|variant| {
         let name = &variant.name;
         let value = &variant.value;
         quote!(#value => Self::#name,)
     });
 
+    let maybe_default = (!raw.variants.iter().any(|v| v.attrs.default)).then(|| quote!(#[default]));
     quote! {
         #( #docs )*
-        #[derive(Clone, Copy, Debug, PartialEq, Eq)]
+        #[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
         #[repr(#typ)]
         pub enum #name {
             #( #variants )*
             #[doc(hidden)]
+            #maybe_default
```

Comment:
I don't think we ever want to have `Unknown` be the default. The existence of `Unknown` is sort of a hack; it is working around the fact that it's possible for an enum value to be invalid in an existing font, and we need some way to represent that, but we cannot compile the `Unknown` variant, and we don't want the user to be constructing them.

I think that if the user does not specify a `#[default]` variant, we should not implement the trait, and the user can figure out what they want to do.

Reply:
That's fair, but then we are left with the issue that you get an errors whose solution is less than obvious when you generate a broken compile type. Maybe I'll file an issue to have a nice error in that case.